### PR TITLE
SUS-3076 | WikiaNewtalk: introduce wfBuildNewtalkWhereCondition()

### PR DESCRIPTION
**Pull request by macbre**
Return a WHERE condition that will match either entries for an anon (by IP address) or a user account (by user ID) and use only these columns.

https://wikia-inc.atlassian.net/browse/SUS-3076

## `WHERE` conditions

Generated by `wfBuildNewtalkWhereCondition` helper:

```php
> var_dump( wfBuildNewtalkWhereCondition( $u, true ) )
array(2) {
  'sn_wiki' =>
  string(8) "plpoznan"
  'sn_user_id' =>
  int(119245)
}

> var_dump( wfBuildNewtalkWhereCondition( $u ) )
array(1) {
  'sn_user_id' =>
  int(119245)
}
```

```php
> var_dump( wfBuildNewtalkWhereCondition( $wgUser ) )
array(1) {
  'sn_user_ip' =>
  string(9) "127.0.0.1"
}

> var_dump( wfBuildNewtalkWhereCondition( $wgUser, true ) )
array(2) {
  'sn_wiki' =>
  string(8) "plpoznan"
  'sn_user_ip' =>
  string(9) "127.0.0.1"
}
```

## Example queries

Two indices were added:

```sql
ALTER TABLE `shared_newtalks` ADD INDEX `idx_user_ip_wiki` (sn_user_ip, sn_wiki), ADD INDEX `idx_user_id_wiki` (sn_user_id, sn_wiki);
```

```sql
mysql@geo-db-sharedb-master.query.consul[wikicities]>explain select count(*) from shared_newtalks where sn_user_ip = '127.0.0.1';
+----+-------------+-----------------+------+------------------+------------------+---------+-------+------+--------------------------+
| id | select_type | table           | type | possible_keys    | key              | key_len | ref   | rows | Extra                    |
+----+-------------+-----------------+------+------------------+------------------+---------+-------+------+--------------------------+
|  1 | SIMPLE      | shared_newtalks | ref  | idx_user_ip_wiki | idx_user_ip_wiki | 258     | const |    1 | Using where; Using index |
+----+-------------+-----------------+------+------------------+------------------+---------+-------+------+--------------------------+
1 row in set (0.00 sec)

mysql@geo-db-sharedb-master.query.consul[wikicities]>explain select count(*) from shared_newtalks where sn_wiki = 'muppet' and sn_user_id = 8;
+----+-------------+-----------------+------+------------------+------------------+---------+-------------+------+--------------------------+
| id | select_type | table           | type | possible_keys    | key              | key_len | ref         | rows | Extra                    |
+----+-------------+-----------------+------+------------------+------------------+---------+-------------+------+--------------------------+
|  1 | SIMPLE      | shared_newtalks | ref  | idx_user_id_wiki | idx_user_id_wiki | 39      | const,const |    1 | Using where; Using index |
+----+-------------+-----------------+------+------------------+------------------+---------+-------------+------+--------------------------+
1 row in set (0.00 sec)

mysql@geo-db-sharedb-master.query.consul[wikicities]>explain select count(*) from shared_newtalks where sn_user_id = 8;
+----+-------------+-----------------+------+------------------+------------------+---------+-------+------+-------------+
| id | select_type | table           | type | possible_keys    | key              | key_len | ref   | rows | Extra       |
+----+-------------+-----------------+------+------------------+------------------+---------+-------+------+-------------+
|  1 | SIMPLE      | shared_newtalks | ref  | idx_user_id_wiki | idx_user_id_wiki | 5       | const |    1 | Using index |
+----+-------------+-----------------+------+------------------+------------------+---------+-------+------+-------------+
1 row in set (0.00 sec)
```

### CODE REVIEW

File: extensions/wikia/WikiaNewtalk/wikia_newtalk.php [hunk 1]
```diff
@@ -130,6 +140,7 @@ function wfGetWikiaNewtalk( User $user, &$talks ) {
 				__METHOD__
 			);
 			while( $row = $dbr->fetchObject( $sth ) ) {
+				// TODO: use GlobalTitle
```
Comment (WikiaTech) [inline extensions/wikia/WikiaNewtalk/wikia_newtalk.php:143]: INFO Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 1 issue
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

File: extensions/wikia/WikiaNewtalk/wikia_newtalk.php [hunk 2]
```diff
@@ -15,6 +15,27 @@ function wfWikiaNewTalkMemcKey( User $user ) {
...
+ * @return array
+ */
+function wfBuildNewtalkWhereCondition( User $user, $includeWikiId = false ) : array {
+	$wikiCondition = $includeWikiId ? [ 'sn_wiki' => wfWikiID() ] : [];
+
+	if ( $user->isAnon() ) {
+		return array_merge( $wikiCondition, [ 'sn_user_ip' => $user->getName() ] );
+	}
+	else {
+		return array_merge( $wikiCondition, [ 'sn_user_id' => $user->getID() ] );
```
Comment (mszabo-wikia) [inline extensions/wikia/WikiaNewtalk/wikia_newtalk.php:34]: No need for `else`

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

**Review by JamesHawking — APPROVED** [review]

**Review by ghost — COMMENTED** [review]
On the contrary I like the `else` :-)  I don't like multiple `return` but… I am academic :)

**Review by ghost — APPROVED** [review]

File: extensions/wikia/WikiaNewtalk/wikia_newtalk.php (hunk 2 above)
Comment (macbre) [reply extensions/wikia/WikiaNewtalk/wikia_newtalk.php:34]: I second Mix opinion - `else` makes logical sense here. If it's an anon do foo, else do bar.

**Review by mszabo-wikia — APPROVED** [review]

**Pull request merged**